src/controllers/users.controller.ts

```diff
@@ -23,27 +23,27 @@ export class UserController {
 
   @Authorized()
   @Get()
-  index(): Promise<User[]> {
-    return this.userRepository.find();
+  async index(): Promise<User[]> {
+    return await this.userRepository.find();
   }
 
   @Get('/:id')
-  show(@Param('id') id: number): Promise<User | undefined> {
-    return this.userRepository.findOne(id);
+  async show(@Param('id') id: number): Promise<User | undefined> {
+    return await this.userRepository.findOne(id);
   }
 
   @Post()
-  post(@Body() user: any): Promise<InsertResult> {
-    return this.userRepository.insert(user);
+  async post(@Body() user: any): Promise<InsertResult> {
+    return await this.userRepository.insert(user);
   }
 
   @Put('/:id')
-  put(@Param('id') id: number, @Body() user: any): Promise<UpdateResult> {
-    return this.userRepository.update(id, user);
+  async put(@Param('id') id: number, @Body() user: any): Promise<UpdateResult> {
```

Comment:
Not sure about the returns here... we may want to have a base controller with custom responses or something like that.... errors should be handled by routing-controller package.